### PR TITLE
Validation of combined fragments

### PR DESCRIPTION
**Pull request by georgiana-ecobici**
Change-Id: Ib9b0de72c3033a9e6414f5bfd8eb601579fd5df6
Signed-off-by: Georgiana-Elena ECOBICI <georgiana-elena.ecobici@thalesgroup.com>

### CODE REVIEW

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 1]
```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
...
 	}
+	
+	/*
+	 * Expression shall not be empty
+	 */
+	@Check
+	def checkCombinedFragmentEmptyExpression(CombinedFragment combinedFragment) {
+		if (combinedFragment.expression == null || combinedFragment.expression.isEmpty) {
+			error(
+				'Expression can not be empty',
```
Comment (MalinaStefaniaStoicanescu) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:277]: for state fragments, if an element is missing (null in this case), the validation sounds something like "please insert .. ". we could use the same pattern or we could change for state fragments :-?

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 2]
```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
...
+	/*
+	 * Expression shall not be empty
+	 */
+	@Check
+	def checkOperandEmptyExpression(Operand operand) {
+		if (operand.expression == null || operand.expression.isEmpty) {
+			error(
+				'Expression can not be empty',
+				TextualScenarioPackage.Literals.OPERAND__EXPRESSION
+			)
```
Comment (MalinaStefaniaStoicanescu) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:292]: same

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 3]
```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
...
+						TextualScenarioPackage.Literals.OPERAND__KEYWORD
+					)
+				}
+			}
+		}
+	}
+
+	/*
+	 * No keyword shall be put on a combined fragment that is not ALT
+	 */
```
Comment (MalinaStefaniaStoicanescu) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:334]: this description doesn't match the method

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend (hunk 1 above)
Comment (georgiana-ecobici) [reply plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:277]: The expression is a String, is not a Capella object, to me, "expression can not be empty" sonds more appropriate that "insert...".

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend [hunk 4]
```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
...
+			}
+			if (!EmbeddedEditorInstanceHelper.checkValidTimeline(timeline)) {
+				unexistingTimelines += " " + timeline;
+			}
+		}
+		if (!undefinedTimelines.isEmpty) {
+			error('Timelines not defined in text:' + undefinedTimelines,
+				TextualScenarioPackage.eINSTANCE.getCombinedFragment_Timelines())
+		}
+		if (!unexistingTimelines.isEmpty) {
```
Comment (MalinaStefaniaStoicanescu) [inline plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:348]: for this if's, you could change the order. if a timeline doesn't exist, it is not defined in xtext too.
in this case, the error displayed will be 'Timelines not defined in text..', but the real error is that it's not exist. 
Also, both errors will not be displayed in the same time :-?

Comment (georgiana-ecobici) [reply plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:348]: done, changed the order.

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend (hunk 1 above)
Comment (MalinaStefaniaStoicanescu) [reply plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:277]: okay, we could change the errors for state fragments in this case

File: plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend (hunk 3 above)
Comment (georgiana-ecobici) [reply plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend:334]: Done

**Review by MalinaStefaniaStoicanescu — APPROVED** [review]

**Pull request merged**